internal/adapters/operation/operation_storage_redis_test.go

```diff
@@ -876,3 +876,140 @@ func TestCleanOperationsHistory(t *testing.T) {
+				DefinitionName: "test-definition",
+				CreatedAt:      now,
+				Input:          []byte("hello test"),
+				ExecutionHistory: []operation.OperationEvent{
+					{
+						CreatedAt: time.Date(1999, time.November, 19, 6, 12, 15, 0, time.UTC),
+						Event:     "some-event",
+					},
+				},
+			},
+			{
+				ID:             "some-op-id-2",
+				SchedulerName:  schedulerName,
+				Status:         operation.StatusFinished,
+				DefinitionName: "test-definition",
+				CreatedAt:      now.Add(-23 * time.Hour),
+				Input:          []byte("hello test"),
+				ExecutionHistory: []operation.OperationEvent{
+					{
+						CreatedAt: time.Date(1999, time.November, 19, 6, 12, 15, 0, time.UTC),
+						Event:     "some-event",
+					},
+				},
+			},
+			{
+				ID:             "some-op-id-3",
+				SchedulerName:  schedulerName,
+				Status:         operation.StatusFinished,
+				DefinitionName: "test-definition",
+				CreatedAt:      now.Add(-25 * time.Hour),
+				Input:          []byte("hello test"),
+				ExecutionHistory: []operation.OperationEvent{
+					{
+						CreatedAt: time.Date(1999, time.November, 19, 6, 12, 15, 0, time.UTC),
+						Event:     "some-event",
+					},
+				},
+			},
+			{
+				ID:             "some-op-id-4",
+				SchedulerName:  schedulerName,
+				Status:         operation.StatusFinished,
+				DefinitionName: "test-definition",
+				CreatedAt:      now.Add(-29 * time.Hour),
+				Input:          []byte("hello test"),
+				ExecutionHistory: []operation.OperationEvent{
+					{
+						CreatedAt: time.Date(1999, time.November, 19, 6, 12, 15, 0, time.UTC),
+						Event:     "some-event",
+					},
+				},
+			},
+		}
+
+		t.Run("if operations history is not empty it deletes all expired operations", func(t *testing.T) {
+			client := test.GetRedisConnection(t, redisAddress)
+			clock := clockmock.NewFakeClock(time.Now())
+			operationsTTlMap := map[Definition]time.Duration{}
+			storage := NewRedisOperationStorage(client, clock, operationsTTlMap)
+			expectedOperations := []*operation.Operation{operations[1], operations[0]}
+
+			for _, op := range operations {
+				executionHistoryJson, err := json.Marshal(op.ExecutionHistory)
+				require.NoError(t, err)
+
+				err = client.ZAdd(context.Background(), storage.buildSchedulerHistoryOperationsKey(op.SchedulerName), &redis.Z{
+					Member: op.ID,
+					Score:  float64(op.CreatedAt.Unix()),
+				}).Err()
+				require.NoError(t, err)
+
+				err = client.HSet(context.Background(), storage.buildSchedulerOperationKey(op.SchedulerName, op.ID), map[string]interface{}{
+					idRedisKey:                 op.ID,
+					schedulerNameRedisKey:      op.SchedulerName,
+					statusRedisKey:             strconv.Itoa(int(op.Status)),
+					definitionNameRedisKey:     op.DefinitionName,
+					createdAtRedisKey:          op.CreatedAt.Format(time.RFC3339Nano),
+					definitionContentsRedisKey: op.Input,
+					executionHistoryRedisKey:   executionHistoryJson,
+				}).Err()
+				require.NoError(t, err)
```

Comment:
@mflilian, we are adding all operations to the history, and creating all operations hashMaps here, this means that none of these operations will be considered expired since all of them exists, this test is not doing what it says.

In this scenario, we should add all operations IDs to the history, but only create it's hashMaps for, let's say two operations (e.g operations[0] and operations[1]), that way operations[2] and operations[3] would be considered as expired since they don't exist as hashMaps in the redis, but are contained in the history.